### PR TITLE
Add a new reranking based on mosec.

### PR DESCRIPTION
**Pull request by miaojinc**
## Description

Add a new reranking based on mosec.

## Issues

`n/a`.

## Type of change

List the type of change like below. Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ *] New feature (non-breaking change which adds new functionality)
- [ ] Breaking change (fix or feature that would break existing design and interface)

## Dependencies

mosec
ipex

## Tests

see the new added test script: tests/test_embeddings_langchain-mosec.sh

### CODE REVIEW

**Review by XinyuYe-Intel — APPROVED** [review]

File: comps/reranks/langchain-mosec/mosec-docker/server-ipex.py [hunk 1]
```diff
@@ -0,0 +1,196 @@
+# Copyright 2024 MOSEC Authors
```
Comment (ftian1) [inline comps/reranks/langchain-mosec/mosec-docker/server-ipex.py:1]: can you pls check with @chensuyue if this license is correct?

Comment (miaojinc) [reply comps/reranks/langchain-mosec/mosec-docker/server-ipex.py:1]: > can you pls check with @chensuyue if this license is correct?

Hi Feng,  I will clean up the code.

Comment (chensuyue) [reply comps/reranks/langchain-mosec/mosec-docker/server-ipex.py]: Yes, please keep the 3rd party copyright.

Comment (miaojinc) [reply comps/reranks/langchain-mosec/mosec-docker/server-ipex.py]: OK, I will keep the 3rd party copyright, and re-push the commit. Thanks for the confirmation.

**Review by ftian1 — APPROVED** [review]

**Pull request merged**